src/jinja2/idtracking.py

```diff
@@ -118,23 +118,19 @@ def load(self, name: str) -> None:
             self._define_ref(name, load=(VAR_LOAD_RESOLVE, name))
 
     def branch_update(self, branch_symbols: t.Sequence["Symbols"]) -> None:
-        stores: t.Dict[str, int] = {}
+        stores: t.Set[str] = set()
         for branch in branch_symbols:
-            for target in branch.stores:
-                if target in self.stores:
-                    continue
-                stores[target] = stores.get(target, 0) + 1
+            stores.update(branch.stores)
+
+        stores.difference_update(self.stores)
```

Comment:
The change from checking `if target in self.stores: continue` to using `difference_update` on two sets performs roughly the same goal: Making sure we don't override the loading behaviour of a symbol already defined in this context.